static/js/query-configs/website_collections.ts

```diff
@@ -136,3 +150,94 @@ export const createWebsiteCollectionMutation = (
     })
   }
 })
+
+export type WebsiteCollectionItemListing = Record<
+  number,
+  WebsiteCollectionItem[]
+>
+
+export const websiteCollectionItemsRequest = (id: number): QueryConfig => {
+  return {
+    url:       wcItemsApiUrl.param({ collectionId: id }).toString(),
+    transform: (body: WebsiteCollectionItem[]) => ({
+      websiteCollectionItems: {
+        [id]: body
+      }
+    }),
+    update: {
+      websiteCollectionItems: mergeRight
+    }
+  }
+}
+
+export const createWebsiteCollectionItemMutation = (
+  item: WCItemCreateFormFields,
+  collectionId: number
+): QueryConfig => ({
+  url:       wcItemsApiUrl.param({ collectionId }).toString(),
+  body:      item,
+  transform: (body: WebsiteCollectionItem) => ({
+    websiteCollectionItems: {
+      [collectionId]: [body]
+    }
+  }),
+  update: {
+    websiteCollectionItems: (
+      prev: WebsiteCollectionItemListing,
+      next: WebsiteCollectionItemListing
+    ) => ({
+      [collectionId]: [...prev[collectionId], ...next[collectionId]]
+    })
+  },
+  options: {
+    method:  "POST",
+    headers: {
+      "X-CSRFTOKEN": getCookie("csrftoken") || ""
+    }
+  }
+})
+
+/**
+ * This mutation is for changing the position of
+ * WebsiteCollectionItems.
+ */
+export const editWebsiteCollectionItemMutation = (
+  item: WCItemMoveFormFields,
+  collectionId: number,
+  itemId: number
+): QueryConfig => ({
+  url: wcItemsApiDetailUrl
+    .param({
+      collectionId,
+      itemId
+    })
+    .toString(),
+  body:      item,
+  transform: (body: WebsiteCollectionItem) => ({
+    websiteCollectionItems: body
+  }),
+  update: {
+    websiteCollectionItems: (
+      prev: WebsiteCollectionItemListing,
+      next: WebsiteCollectionItem
+    ) => {
+      const oldIndex = findIndex(
+        item => item.id === next.id,
+        prev[collectionId]
+      )
+
+      const newArray = [...prev[collectionId]]
+      newArray[oldIndex] = next
+
+      return {
+        [collectionId]: arrayMove(newArray, oldIndex, next.position)
```

Comment:
Do we need to adjust the `position` of any of the other items?

Reply:
hmm we could if we wanted to make sure we have full consistency on the frontend, but I don't think it's strictly necessary. We never read, on the frontend, from the `position` attribute to determine what order to display things, and the logic here and in the `handleDrag` function in `WebsiteCollectionItemsEditor` works based on the position of the items in the array rather than the `position` attribute. Another way to say that, is that we're only really using the `position` attribute when we're doing a `PATCH` on one of our existing records.